executor/reduce_executor.rs

```diff
@@ -693,3 +738,46 @@ impl ReducerAPI for StdDoubleExecutor {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+struct StdDecimalExecutor {
+    sum: Decimal,
+    sum_squares: Decimal,
+    count: u64,
+    target: VariablePosition,
+}
+
+impl StdDecimalExecutor {
+    fn new(target: VariablePosition) -> Self {
+        Self { sum: Decimal::default(), sum_squares: Decimal::default(), count: 0, target }
+    }
+}
+
+impl ReducerAPI for StdDecimalExecutor {
+    fn accept<Snapshot: ReadableSnapshot>(
+        &mut self,
+        row: &MaybeOwnedRow<'_>,
+        context: &ExecutionContext<Snapshot>,
+        storage_counters: StorageCounters,
+    ) {
+        if let Some(value) = extract_value(row, self.target, context, storage_counters) {
+            let unwrapped = value.unwrap_decimal();
+            self.sum_squares += (unwrapped * unwrapped) * row.multiplicity();
+            self.sum += unwrapped * row.multiplicity();
+            self.count += row.multiplicity();
+        }
+    }
+
+    fn finalise(self) -> Option<VariableValue<'static>> {
+        if self.count > 1 {
+            let sum = self.sum;
+            let sum_squares = self.sum_squares;
+            let n = self.count;
+            let mean = sum / n;
+            let sample_variance: Decimal = (sum_squares + n * mean * mean - 2 * sum * mean) / (n - 1);
+            Some(VariableValue::Value(Value::Double(sample_variance.to_f64().sqrt())))
```

Comment:
`sqrt` is the only reason `StdDecimal` returns `Double` instead of `Decimal`. The question remains whether it's even sensible to have `sqrt(Decimal) -> Decimal` in the first place. I'm inclined to say that once we're going irrational, we should probably not overpromise the precision.